jest.config.js

```diff
@@ -41,6 +41,9 @@ module.exports = {
     platforms: ['ios', 'android'],
   },
   moduleNameMapper: {
+    // These mappers allow out-of-tree platforms tests to seamlessly resolve RN imports
+    '^react-native/(.*)': '<rootDir>/packages/react-native/$1',
+    '^react-native$': '<rootDir>/packages/react-native/index.js',
```

Comment:
Oh hey I remember hitting this back in the day for RN Windows.

Something else you will run into is resolving files correctly based on platform extension, if you end up adding `visionos` there (even in internals of the fork).

@tido64 you ended up building a custom resolve for this, or similar right?  Maybe we should upstream the solution we had.

Reply:
We just upstreamed one bugfix (custom resolvers were not overridable when OOT was defined) and create a custom resolver here: https://github.com/callstack/react-native-visionos/pull/32

We chose not to change the `Platform` and instead do something custom, because there's so much overlap between visionOS and iOS and we want to leverage 3rd party libs that rely on `ios` platform

Reply:
A nuance is that Jest uses a different resolver from Metro, but IIRC had some declarative fields to customize resolution. I looked it up, and here is the MSFT one I think: https://github.com/microsoft/rnx-kit/blob/main/packages/jest-preset/src/index.js

Reply:
As @NickGerleman pointed out, it's built into our Jest preset. `react-native-windows` uses it extensively: https://github.com/search?q=repo%3Amicrosoft%2Freact-native-windows%20%40rnx-kit%2Fjest-preset&type=code

It should work for `react-native-visionos` as well. If it doesn't, let me know.